ui/phase_manager.py

```
@@ -0,0 +1,344 @@
+
+        self._active_phase = None
+        self._active_step = None
+
+        self._queue.publish({
+            "type": "phase_done",
+            "phase": phase,
+            "progress": self._get_progress(),
+            "queue": self._get_pending(),
+        })
+
+    def error_phase(self, phase_id: str, error: str, name: str = None) -> None:
```

Ruff (0.15.9)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)